docs/source/tutorial.rst

```diff
@@ -336,33 +335,96 @@ You can find the team color by subscribing to the relevant topic (this should
+
+    - `<https://www.journaldev.com/44327/arrow-operator-c-plus-plus>`_
+    - `<https://www.geeksforgeeks.org/arrow-operator-in-c-c-with-examples/>`_
+
+* Notice the docstrings for the radio header file. It explains that it's the abstract superclass of the network_radio and sim_radio node. If you are unfamiliar with the concept of abstraction, `here <https://www.pythontutorial.net/python-oop/python-abstract-class/>` is more information. When exploring these two nodes, you will discover a third file named "node"_main.cpp which contains the main function for its respective node. This structure is intended to make writing the CMakeLists.txt file for the directory easier.
+
+While those files give you a peek into structuring your node files, they don't give you insight into how our build system works. When your compiler is building our code base, `CMake <https://cmake.org/overview/>`_ allows you to forgo the need to maintain settings specific to your compiler/build environment. As a result, you need to add your node to the appropriate CMake files.
+
+CMakeLists.txt files are used to make standard build files for the directory. It locates files, libraries, and executables to support complex directory hierarchies. Locate the CMakeLists.txt file in ``robocup-software/soccer/src/soccer``.
+
+**Key Sections**
+
+* Notice all the source files in the ``ROBOCUP_LIB_SRC``. You will find the radio files that you explored earlier along with other crucial source files that aid in motion control, soccer ui, etc. You will get familiar with these files as you continue to contribute to the codebase.
+* Many of the nodes have an environment variable set for their "node"_main.cpp. (HINT: There is a connection between the nodes that have these variables and the nodes that are returned by the ROS CLI)
+* Each of the nodes that have a source variable are added in the CMake file as `target_sources`
+
+You may have started noticing the pattern of adding to the key sections of the CMake files. You need to find the parts of the file where the other nodes are located and follow the format for your own node. While it is important to understand the meaning of each section, you will eventually learn as you gain experience. As explained earlier, CMake files are build files for the directory. So check if you need to make additions to the ``robocup-software/soccer/src`` and ``robocup-software/soccer`` CMake files. Follow the pattern mentioned above.
+
+You're almost there! One more step to go before acquiring all the necessities to create your own node. Launch files in ROS are a convenient way of starting up multiple nodes, setting initial parameters, and other requirements. Find the launch directory and explore the file that would pertain to where your node would be located. As you already know, follow the pattern for the other nodes that are functionally similar to your node.
```

Comment:
```suggestion
You're almost there! One final file you have to create is a launch file. Launch files in ROS are a convenient way of starting up multiple nodes, setting initial parameters, and other requirements. Find the launch directory and explore the file that would pertain to where your node would be located. As you already know, follow the pattern for the other nodes that are functionally similar to your node.
```
which launch directory? specific file path would make this easier

Reply:
and or specific nodes to look at launch files for

Reply:
I will add where the launch files are located and guide them to the particular file. However, telling them which node to copy the format for is unnecessary because most of the nodes follow the same format. Common sense would tell them to follow what most of the nodes do.